### ISSUE TITLE
Language should use the Language Tags Vocabulary

### ISSUE THREAD

**Issue by gsergiu** [w3c/web-annotation#134]
Currently the definition of language property 
http://www.w3.org/TR/annotation-model/#external-resources
points the the specification of how the values should be formated. 
https://tools.ietf.org/html/rfc5646

however the definition should reference the existing vocabulary as well ..
see: 
http://www.w3.org/International/articles/language-tags/
and 
http://www.iana.org/assignments/language-subtag-registry/language-subtag-registry

**Comment by gsergiu** [comment]
this is similar an can be solved together with #133

**Comment by azaroth42** [comment]
As per comment on #133, those documents are also not good normative references. Again, happy to be corrected :)

**Comment by gsergiu** [comment]
Well … I think that the trick with “should” and “must” is used enough in the document to leave the standard open for extension, even if in 90+% percent of cases  a standard representation will be used.
I don’t know which are the mechanisms to separate the normative from non-normative part of the standard. But I think that at list of “non normative” Note would be more than welcome for those who implement the standard.

BR,
Sergiu

Von: Rob Sanderson [mailto:notifications@github.com]
Gesendet: Dienstag, 12. Januar 2016 17:53
An: w3c/web-annotation
Cc: Gordea Sergiu
Betreff: Re: [web-annotation] Language should use the Language Tags Vocabulary (#134)

As per comment on #133https://github.com/w3c/web-annotation/issues/133, those documents are also not good normative references. Again, happy to be corrected :)

**Comment by iherman** [comment]
Let us use the same approach as in #133, with the IANA address taking the role of the vocabulary list. We should probably call this one out explicitly when asking for an I18N review.

**Comment by iherman** [comment]
Decision on 2016-02-05,  http://www.w3.org/2016/02/05-annotation-irc#T16-42-38-1

Use http://www.w3.org/International/articles/language-tags/ as a reference

**Comment by gsergiu** [comment]
great thanks!

**Comment by hugomanguinhas** [comment]
Hello all,

Given the resolution of this issue, I was wondering if there are strong reasons why the specification does not allow a literal to be annotated with the language (i.e. both dc:language and xml:lang refer to http://tools.ietf.org/html/bcp47 as normative spec)?

This is the normal practice when representing data in RDF and the use of dc:language is normally used to state the language of the resource being described and not the language of the terms being used to describe it.

BR, Hugo

**Comment by iherman** [comment]
> Given the resolution of this issue, I was wondering if there are strong reasons why the specification does not allow a literal to be annotated with the language (i.e. both dc:language and xml:lang refer to http://tools.ietf.org/html/bcp47 http://tools.ietf.org/html/bcp47 as normative spec)?
> This is the normal practice when representing data in RDF and the use of dc:language is normally used to state the language of the resource being described and not the language of the terms being used to describe it.

Not sure what you mean. Do you mean that we should have a selector which includes a language tag, too? Is there a use case for this?

**Comment by hugomanguinhas** [comment]
Hi Ivan,

I was talking about the specification not allowing Literals to be "associated" with language tags as defined in https://www.w3.org/TR/rdf11-primer/#section-literal... so I would expect an annotation like the following to be alright: 

``` json-ld
{
  "motivation": "commenting",
  "body": { "@value": "my useful comment", "@language": "en" },
  "target": "http://..."
}
```

...unfortunatelly, there isn't a more compact syntax like in XML/RDF to express the language.

**Comment by gsergiu** [comment]
well ... in this example we prevent users to use @ in their text. 
If something like this should be supported in jsonld the escaping of @ is necesary.
instead of supporting "comment@@en", I think that the following representation is more machine and user friendly:
{
"body":"comment",
"language":"en"
}

(and natively supported by json parsers)

**Comment by iherman** [comment]
@hugomanguinhas I still miss something. Isn't it exactly what

https://w3c.github.io/web-annotation/model/wd/#h-textual-body

gives you? Your example above would be:

``` json-ld
{
  "motivation": "commenting",
  "body": { 
    "text": "my useful comment",
    "language": "en" 
  },
  "target": "http://..."
}
```

Sorry if I still miss something.

`comment@en` or `comment@@en` would not be legal JSON-LD, and I do not think we should introduce some sort of a microsyntax into the JSON dialect we define.

**Comment by hugomanguinhas** [comment]
Hi Ivan and Sergiu,

Just to clarify that I am not proposing a new syntax, I am just questioning why the specification does not allow for Language Literals as they are defined in RDF.

About JSON-LD, even though they look the same there is a significant difference between the two... In my example the Body is a Language rdf:Literal while in Ivan's example it is a rdf:Resource (of type oa:TextualBody) with two properties (dc:language with the language as value and oa:text with the text).

**Comment by gsergiu** [comment]
well ... I expressed my opinion before, that the WA standard should be technology independent. 
The provided examples are non-normative, and one can notice big differences between the rdf/xml and jsonld examples. 
(I'm not very interested in the rdf/xml serialization as I don't use it for now, but I'm interested in the json-ld)

So I asume that by rdf:Literal you related to the Value Objects/ language-tagged string as specified in the jsonld, which define the @value and @language properties 
see
https://www.w3.org/TR/2014/REC-rdf11-concepts-20140225/#dfn-literal-value
https://www.w3.org/TR/json-ld/#value-objects  
https://www.w3.org/TR/json-ld/#dfn-language-tagged-string

So .. if I understood correctly your claim, you want to say that:
1. rdf:Literal is equivalent to language-tagged-string and the representations should be used consistently
2. The bodies should provide support for correct representation of "string literals"/"language-tagged strings"

... in generally I support these claims only that there are some related aspects to take in account 
a. I find "text" to be more appropriate that "value" to represent the free text provided by user. 
b. there must be a clear differentiation between the "value", "text" and "@id"(/httpuri) properties. I think that the "value" is by default ambiguous. So if we want to introduce @value & @language, i would be tempted to propose the followign representation as being correct, but this is a quire verbose one:
{
  "motivation": "commenting",
  "body": { 
    "text": { "@value": "my useful comment",  "@language": "en" }
  },
  "target": "http://..."
}
c. I think that this  "string literals"/"language-tagged strings" issue is actually ... nothing different than the support for "i18n". Which is not included yet in the WA draft. And yes, I do agree this is needed, and it should be considered  first when serializing language-tagged texts.

Personal conclusion:
I hope that I got it right this time, and I didn't bring more confusion in the thread. 
If my understanding was correct, I would opt for moving the "todays discussions" in this thread to own issue, as it diverges from the initial text of the isues and the resolution (proposing usage of html/xml language tags) doesn't conflicts with the issue discussed here.

Br,
Sergiu
